sqlglot/dialects/redshift.py

```diff
@@ -136,11 +136,11 @@ def _parse_query_modifiers(
                     refs.add(
                         (
                             this.args["from"] if i == 0 else this.args["joins"][i - 1]
-                        ).alias_or_name.lower()
+                        ).this.alias.lower()
```

Comment:
I believe we need aliases for the unnesting to work in Redshift. I've found an SO [response](https://stackoverflow.com/a/72870096) that seems to suggest this, but also Redshift's [docs](https://docs.aws.amazon.com/redshift/latest/dg/query-super.html#unnest) only show examples with aliases.

Some queries I tested also back this up:

```sql
-- works
with input_data as (select array(1, 2,  3) as id)
select id2 from input_data as ids, ids.id as id2

-- ERROR: column "id2" does not exist in input_data, id2
with input_data as (select array(1, 2,  3) as id)
select id2 from input_data, input_data.id as id2
```